### DISCUSSION TITLE
How to load PDF and other Docs

### DISCUSSION THREAD

**Discussion by abrahimzaman360** [safevideo/autollm#93] [Q&A] [answered]
How to load PDF and other documents, here are just told about .md

**Comment by talhasarit** [comment, marked as answer]
Hey @AbrahimZaman360,

Thanks for bringing this up! I understand that our documentation primarily mentions `.md` files, which might lead to some confusion.

The `read_files_as_documents` and `read_github_repo_as_documents` functions were designed to be versatile. By default, they support various file types, not just `.md`. The `required_exts` parameter lets users filter the types of files they want to read, but it's entirely optional.

Here's how you can use it:
1. If you want to read all supported file types, simply omit the `required_exts` parameter.
2. If you want to specify file types, include them in the `required_exts` parameter.

For example:
```python
read_files_as_documents(input_dir="your/path/to/documents", required_exts=['.pdf', '.txt'])
```

You can check all our [supported file types here](https://github.com/run-llama/llama_index/blob/main/llama_index/readers/file/base.py#L19-L34).

I hope this clarifies things! If you have more questions or suggestions, please let us know. We appreciate your feedback!
